### PR TITLE
JavaLexed unit tests

### PR DESCRIPTION
**Pull request by Michael1993**
Added tests to show expected behaviour and improve coverage.

### CODE REVIEW

File: javageci-tools/src/test/java/javax0/geci/lexeger/TestJavaLexed.java [hunk 1]
```diff
@@ -17,6 +18,15 @@ void testNullTransformation(List<String> lines) {
         Assertions.assertEquals(expected, result);
     }
 
+    private String buildLexemeTree(JavaLexed sut) {
```
Comment (verhas) [inline javageci-tools/src/test/java/javax0/geci/lexeger/TestJavaLexed.java]: `buildLexemeTree()` is not a good name. It is not a tree. It is a sequential string representation of the list of lexical elements. I would use the method name `toString(JavaLexed javaLexed)`

[the remark this answers is not in the document]
Comment (Michael1993) [reply javageci-tools/src/test/java/javax0/geci/lexeger/TestJavaLexed.java]: Renamed.

File: javageci-tools/src/test/java/javax0/geci/lexeger/TestJavaLexed.java [hunk 2]
```diff
@@ -47,12 +57,9 @@ void iterateThrough() {
         final var source = new TestSource(List.of(
             "private final var apple= \"appleee\";   ",
             "    private final final 13 'aaaaa' "));
-        final var sb = new StringBuilder();
+        String lexed;
```
[the remark this answers is not in the document]
Comment (Michael1993) [reply javageci-tools/src/test/java/javax0/geci/lexeger/TestJavaLexed.java]: Done.

File: javageci-tools/src/test/java/javax0/geci/lexeger/TestJavaLexed.java [hunk 3]
```diff
@@ -76,6 +83,93 @@ void iterateThrough() {
...
+    @DisplayName("Can't remove range with bigger start than end.")
+    void removeRangeThrowsExceptionWhenWrongStartEndOrder() {
+        final var source = new TestSource(List.of(""));
+        try(final var sut = new JavaLexed(source)) {
+            Assertions.assertThrows(IllegalArgumentException.class, () -> sut.removeRange(2, 1));
+        }
+    }
+
+    @Test
+    @DisplayName("Can remove a single lexical element.")
```
[the remark this answers is not in the document]
Comment (Michael1993) [reply javageci-tools/src/test/java/javax0/geci/lexeger/TestJavaLexed.java]: No significance. Now it's removed from the middle.

File: javageci-tools/src/test/java/javax0/geci/lexeger/TestJavaLexed.java [hunk 4]
```diff
@@ -76,6 +83,93 @@ void iterateThrough() {
...
+            lexed = buildLexemeTree(sut);
+        }
+        Assertions.assertEquals("IDENTIFIER[public]\n" +
+            "SPACING[ ]\n" +
+            "IDENTIFIER[static]\n" +
+            "SPACING[ ]\n", lexed);
+    }
+
+    @Test
+    @DisplayName("Can get a single lexical element by id.")
```
[the remark this answers is not in the document]
Comment (Michael1993) [reply javageci-tools/src/test/java/javax0/geci/lexeger/TestJavaLexed.java]: Done.

File: javageci-tools/src/test/java/javax0/geci/lexeger/TestJavaLexed.java (hunk 1 above)
Comment (Michael1993) [reply javageci-tools/src/test/java/javax0/geci/lexeger/TestJavaLexed.java]: Renamed to toLexicalString.

**Pull request merged**